core/src/main/java/org/mule/runtime/core/construct/AbstractPipeline.java

```diff
@@ -236,7 +237,7 @@ protected void doInitialise() throws MuleException {
 
         @Override
         public Event process(final Event event) throws MuleException {
-          if (isTransactionActive()) {
+          if (isTransactionActive() || processingStrategy == SYNCHRONOUS_PROCESSING_STRATEGY_INSTANCE) {
```

Comment:
I think the processingStrategy itself should have a isSynchronous() method. These kind of comparisons are pretty fragile and hard to test.